nsxt/resource_nsxt_host_upgrade_group.go

```diff
@@ -0,0 +1,183 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/upgrade"
+)
+
+func resourceNsxtHostUpgradeGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtHostUpgradeGroupCreate,
+		Read:   resourceNsxtHostUpgradeGroupRead,
+		Update: resourceNsxtHostUpgradeGroupUpdate,
+		Delete: resourceNsxtHostUpgradeGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision":     getRevisionSchema(),
+			"description":  getDescriptionSchema(),
+			"display_name": getDisplayNameSchema(),
+			"tag":          getTagsSchema(),
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Flag to indicate whether upgrade of this group is enabled or not",
+				Optional:    true,
+				Default:     true,
+			},
+			"extended_configuration": {
```

Comment:
I think we can reuse schema for key/value pairs, we already have a func `getSwitchingProfileIdsSchema`, we might want to rename it and generalize the description. Also `advanced_configuration` of edge transport node might benefit from this

Reply:
Switching profile uses a different schema than edge transport node and upgrade - [SwitchingProfileTypeIdEntry](https://vdc-download.vmware.com/vmwb-repository/dcr-public/8e2a4b1a-659f-4be6-a4f4-7ca42b04cb84/9a2c1372-7174-40f0-89d2-27615bfd5d08/api_includes/types_SwitchingProfileTypeIdEntry.html) vs [KeyValuePair](https://vdc-download.vmware.com/vmwb-repository/dcr-public/8e2a4b1a-659f-4be6-a4f4-7ca42b04cb84/9a2c1372-7174-40f0-89d2-27615bfd5d08/api_includes/types_KeyValuePair.html).
I'll consolidate edge and upgrade and leave switching profiles as is.